third_party/move/move-compiler/src/expansion/translate.rs

```diff
@@ -2585,7 +2684,38 @@ fn exp_(context: &mut Context, sp!(loc, pe_): P::Exp) -> E::Exp {
                 EE::BinopExp(exp(context, *pl), op, exp(context, *pr))
             }
         },
-        PE::Borrow(mut_, pr) => EE::Borrow(mut_, exp(context, *pr)),
+        PE::Borrow(mut_, pr) => {
+            if let PE::Index(ref e, ref i) = pr.value {
```

Comment:
Can we pull this out into its own function? Specifically if you use return inside the block its brittle.